tools/dynamodbConnector.py

```diff
@@ -12,63 +12,61 @@ class DynamoDBConnector:
     def __init__(self):
         """Initialise the connection, and create a logger instance."""
         self.logger = logging.getLogger(__name__)
-        self.dynamodb = boto3.resource(
+        self.dynamodb = boto3.client(
             'dynamodb',
-            region_name='eu-west-2',
-            endpoint_url='http://localhost:8765'
         )
         try:
-            if 'scraper_articles' not in self.dynamodb.tables.all():
-                self.logger.error('Article table does not exists.')
+            tables = self.dynamodb.list_tables().get('TableNames')
         except ClientError as e:
             self.logger.error(
                 'Error when initialising the connection [%s]', e
             )
+        else:
+            if 'scraper_articles' not in tables:
+                self.logger.error('Article table does not exists.')
 
     def insert_article(self, file_hash, url):
         """Try to insert an article and its url in the article table. Return
         dynamodb response or `None` if the request fail.
         """
         try:
-            table = self.dynamodb.Table("scraper_articles")
-            response = table.put_item(Item={
-                'file_hash': file_hash,
-                'url': url
-            })
+            return self.dynamodb.put_item(
+                TableName='scraper_articles',
+                Item={
+                    'file_hash': {'S': file_hash},
+                    'url': {'S': url}
+                }
+            )
         except ClientError as e:
             self.logger.error('Couldn\'t insert article [%s]', e)
-            return
-
-        return response
 
     def insert_file_in_catalog(self, file_index, file_path):
         """Insert the newly created file informations into the catalog table.
         """
         try:
-            table = self.dynamodb.Table('scraper_catalog')
-            response = table.put_item(Item={
-                'file_index': file_index,
-                'file_path': file_path,
-                'date_created': datetime.now()
-            })
+            return self.dynamodb.put_item(
+                TableName='scraper_catalog',
+                Item={
+                    'file_index': {'S': file_index},
```

Comment:
why is the item furtner nested? `'file_index': {'S': file_index}` instead of `'file_index': file_index`

Reply:
AWS.

No really, AWS require it this way, we need to specify that `file_index` is a string using the `S` key.